### ISSUE TITLE
Caption on training improvement misleading

### ISSUE THREAD

**Issue by carlogrisetti** [ludwig-ai/ludwig#840]
Hello :)
This piece of code
https://github.com/uber/ludwig/blob/62430e4a0dd7a4fda08d6dcd615fbdbbf53c5377/ludwig/models/trainer.py#L1316-L1318
although procedurally correct (I want to track whether there has been an improvement since the last plateau and learning rate reduction, to look for the next plateau) leads to a technically wrong statement in the CLI output.
Let's assume that the model is training, 5 epochs have passed and the learning rate gets automatically dropped as per the `reduce_learning_rate_on_plateau` parameters. The "last improvement epoch" gets reset to the current epoch (to keep track of the fact that we reduced the learning rate, and we thus need to wait another 5 epochs without improvement before reducing the learning rate again), but in the CLI the next epoch will show that it's only 1 epoch since the last improvement (which should have been 6), the next one will show 2 epochs (but it should be 7), and this only gets fixed as soon as we have an actual improvement.

It's trivial, but maybe it gets to your nerves as much as it does to me :)
